### PR TITLE
Add OwnedSlice::RefRaw to keep track of raw pointers

### PR DESCRIPTION
**Pull request by domenukk**
This should remove UB for map accesses from the Rust side, see #446
Technically, a user can now create a `RefRaw` directly and call the non-unsafe functions, but I think that's okay (or is there any way to restrict enum creation?)

### CODE REVIEW

**Comment by domenukk** [comment]
I might want to do something like this to "hide" the enum for direct creation
https://stackoverflow.com/a/28090996/1345238

File: libafl/src/bolts/ownedref.rs [hunk 1]
```diff
@@ -166,21 +166,65 @@ where
...
+enum OwnedSliceInner<'a, T: 'a + Sized> {
+    /// A ref to a raw slice and length
+    RefRaw(*const T, usize),
     /// A ref to a slice
     Ref(&'a [T]),
     /// A ref to an owned [`Vec`]
     Owned(Vec<T>),
 }
 
+/// The struct to simply wrap OwnedSliceInner to avoid exposing the pointer in it.
```
Comment (domenukk) [inline libafl/src/bolts/ownedref.rs:178]: The OwnedSliceInner is implementation detail and we probably don't want to document it.
I'd just use the original `/// Wrap a slice and convert to a Vec on serialize` comment

File: libafl/src/bolts/ownedref.rs [hunk 2]
```diff
@@ -166,21 +166,65 @@ where
...
+impl<'a, T> From<Vec<T>> for OwnedSlice<'a, T> {
+    fn from(vec: Vec<T>) -> Self {
+        Self {
+            inner: OwnedSliceInner::Owned(vec),
+        }
+    }
+}
+
+/// Create a new [`OwnedSlice`] from a reference to a slice
+impl<'a, T> From<&'a [T]> for OwnedSlice<'a, T> {
```
Comment (domenukk) [inline libafl/src/bolts/ownedref.rs:209]: We should add these for OwnedSliceMut, too, then it's perfect, thanks! <3

**Comment by tokatoka** [comment]
I wonder if we should change every `match self.inner` to `match &self.inner` (for consistency)

**Pull request merged**